### DISCUSSION TITLE
REST API: How to search photos by storage path (curl)?

### DISCUSSION THREAD

**Discussion by ppprism** [photoprism/photoprism#2080] [Q&A] [answered]
Hello,

How can I search all photo UIDs present in a specific storage location/path via REST?

Say, `path` = `my/path` (where `path` of a photo can e.g. be retrieved via `http://localhost:2342/api/v1/photos/<PUID>`).

I tried following:
```sh
curl -Ss -H "X-Session-ID: 1234" "http://localhost:2342/api/v1/photos?path=my/path"
#  or escaped slash
curl -Ss -H "X-Session-ID: 1234" "http://localhost:2342/api/v1/photos?path=my%2Fpath"
```
in analogy to the `count` example from [docs](https://docs.photoprism.app/developer-guide/ui/rest-api/), after having seen [`SearchPhotos`](https://pkg.go.dev/github.com/photoprism/photoprism/internal/api?utm_source=godoc#SearchPhotos) and the  [list of possible parameters](https://github.com/photoprism/photoprism/blob/c256664a1b722487947f48c5c0f69797bf9dbde7/internal/form/search_photos.go). But I did get
```
{"code":400,"error":"Invalid request"}
```
Desired result are all photo UIDs for the given originals storage path, which I then use to add labels to correspnding photos.

Currently REST documentation is quite hard to understand, especially for people not firm with go language. Additional examples in the docs would help hugely.

Appreciate any help for above issue. Thanks!

**Comment by kvalev** [comment, marked as answer]
I think its `photos?filter=path:"my/path"` encoded of course.

**Reply by ppprism** [reply]
Thanks for replying, unfortunately it does not work. Tried
```
curl -Ss -H "X-Session-ID: 1234" "http://localhost:2342/api/v1/photos?filter=path%3A%22my%2Fpath%22"

curl -Ss -G --data-urlencode "filter=path:my/path" -H "X-Session-ID: 1234" "http://localhost:2342/api/v1/photos"
```
Same error message as above

**Reply by kvalev** [reply]
You also need to add the `count` and `offset` parameters. This works for me:

```
curl --location --request GET 'http://localhost:2342/api/v1/photos?count=60&offset=0&filter=path:"/mypath"' \
--header 'X-Session-ID: ...'
```

**Reply by ppprism** [reply]
Worked, great! Obligatory `count` was the missing piece of the puzzle.

Do you know, when we need to use `filter` instead of directly adding the query parameter field, as with `count`? I couldn't find anything related in the docs.

Thank you very much for your help!

**Reply by kvalev** [reply]
No idea to be honest... It is even a bit more confusing, because there is also the `q` parameter, which can also be used to filter by path (the star at the end is the important bit):

```
curl --location --request GET 'http://localhost:2342/api/v1/photos?count=60&offset=0&q=filename:mypath/*' \
--header 'X-Session-ID: ...'
```

**Reply by ppprism** [reply]
hey @kvalev,
posted this question on gitter and  [here](https://gitter.im/browseyourlife/community?at=621e35636b91242320125213)  is what Michael Mayer said:
> filter is for internal use only, e.g. smart albums like moments and calendar
you can use q or individual fields, whatever is best... q will be parsed. Individual fields don't need to be parsed.

I just tried `path` with `q` and even directly as field and it works as long `count` is also included.

**Reply by lastzero** [reply]
- filter is for internal use only, e.g. smart albums like moments and calendar
- you can use q or individual fields, whatever is best... q will be parsed, individual fields don't need to be parsed
